File: my_py_pkg/my_py_pkg/add_two_ints_cl.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from functools import partial
from example_interfaces.srv import AddTwoInts

class AddTwoIntsClient(Node):
    def __init__(self):
        super().__init__('add_two_ints_client')
        self.client = self.create_client(AddTwoInts, 'add_two_ints')
        while not self.client.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('Service not available, waiting again...')
        
        # Call the service with the desired values
        self.call_add_two_ints_server(5, 5)

    def call_add_two_ints_server(self, a, b):
        # Create a request object
        request = AddTwoInts.Request()
        request.a = a
        request.b = b

        # Call the service asynchronously and add the callback
        future = self.client.call_async(request)
        future.add_done_callback(partial(self.callback_call_add_two_ints_cl, a=a, b=b))

    def callback_call_add_two_ints_cl(self, future, a, b):
        try:
            # Get the response from the service call
            response = future.result()
            self.get_logger().info(f'{a} + {b} = {response.sum}')
        except Exception as e:
            self.get_logger().error(f'Service call failed: {e}')

def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/add_two_ints_cl.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class addtwointsclientnode:public rclcpp::Node
{
    public:
    addtwointsclientnode():Node("add_two_ints_client")
    {
        thread1=std::thread(std::bind(&addtwointsclientnode::calladdtwointsservice,this,3,4));
    }

    void calladdtwointsservice(int a,int b)
    {
        auto client = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
        while(!client->wait_for_service(std::chrono::seconds(1)))
        {
            RCLCPP_INFO(this->get_logger(),"Waiting for service to appear...");
        }

        auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
        request->a=a;
        request->b=b;

        auto future = client->async_send_request(request);

        try{
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(),"%d + %d = %ld",a,b,response->sum);
        }
        catch(const std::exception& e)
        {
            RCLCPP_ERROR(this->get_logger(),"Service call failed");
        }
    }

    private:
        std::thread thread1;
};

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<addtwointsclientnode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}